pkg/agent/controller/l7visibility/l7visibility_controller.go

```diff
@@ -0,0 +1,404 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l7visibility
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/config"
+	"antrea.io/antrea/pkg/agent/controller/networkpolicy/l7engine"
+	"antrea.io/antrea/pkg/agent/interfacestore"
+	"antrea.io/antrea/pkg/agent/openflow"
+	"antrea.io/antrea/pkg/agent/types"
+	"antrea.io/antrea/pkg/apis/crd/v1alpha2"
+	"antrea.io/antrea/pkg/util/k8s"
+)
+
+const (
+	controllerName               = "L7VisibilityController"
+	resyncPeriod   time.Duration = 0
+	minRetryDelay                = 5 * time.Second
+	maxRetryDelay                = 300 * time.Second
+	defaultWorkers               = 4
+
+	socketPath = "/var/run/suricata/suricata_eve.socket"
+)
+
+var (
+	errInvalidAnnotation    = fmt.Errorf("annotation key %s can only have values (Ingress/Egress/Both)", types.L7VisibilityAnnotationKey)
+	errPodInterfaceNotFound = fmt.Errorf("interface of Pod not found")
+)
+
+type L7VisibilityController struct {
+	ofClient       openflow.Client
+	interfaceStore interfacestore.InterfaceStore
+
+	podInformer     cache.SharedIndexInformer
+	podLister       corelisters.PodLister
+	podListerSynced cache.InformerSynced
+
+	namespaceInformer     cache.SharedIndexInformer
+	namespaceLister       corelisters.NamespaceLister
+	namespaceListerSynced cache.InformerSynced
+
+	l7Reconciler      *l7engine.Reconciler
+	PodToDirectionMap map[string]v1alpha2.Direction
+	podToDirMapMut    sync.RWMutex
```

Comment:
It's a red flag to have `PodToDirectionMap` be a public member, when it's protected by a mutex. You should make `PodToDirectionMap` private and re-organize code as needed.

s/podToDirMapMut/podToDirectionMapMutex

Reply:
changed to private!